tiledb/sm/query/reader_base.cc

```diff
@@ -726,12 +726,12 @@ ReaderBase::load_tile_chunk_data(
     // dense case).
     if (tile_tuple == nullptr ||
         std::get<0>(*tile_tuple).filtered_buffer().size() == 0) {
-      return {Status::Ok(), nullopt, nullopt, nullopt};
+      return {Status::Ok(), 0, 0, 0};
```

Comment:
Do you know which test this was coming from and where was this exception being thrown?  IIUC the purpose of using `optional` in the return is to indicate whether the buffer exists at all, so returning `0` has a different meaning than returning `nullopt`. This change might be masking another problem such as a missing check(s) for `has_value` before trying to do something with the optional.

Reply:
`Status::Ok()` means that a function is returning some value. `nullopt` is the absence of a value. Line 729 is defective on its face for that reason alone.

Reply:
Exactly.
The exception comes from this line https://github.com/TileDB-Inc/TileDB/blob/dev/tiledb/sm/query/reader_base.cc#L947 that indeed is accessing an `optional` without checking if it `has_value()`. However, it is checking that the returned `Status` is` Ok()`, which is equivalent in our convention. This is they way `optional` return values are checked today in our codebase. If we think we should enhance that with `has_value()` checks we should do that everywhere, but it looks somehow redundant to me.

Reply:
Agree with @ypatia and @eric-hughes-tiledb here. @ihnorton can we merge this and have a separate discussion about checking has_value everywhere (or use *)?

Reply:
If 0s are valid here, then no objection to merging.